src/GitHub.App/ViewModels/GitHubPane/GitHubPaneViewModel.cs

```diff
@@ -410,26 +417,56 @@ async Task UpdateContent(ILocalRepositoryModel repository)
             var isDotCom = HostAddress.IsGitHubDotComUri(repositoryUrl);
             var client = await apiClientFactory.Create(repository.CloneUrl);
             var isEnterprise = isDotCom ? false : await client.IsEnterprise();
+            var notGitHubRepo = true;
 
-            if ((isDotCom || isEnterprise) && await IsValidRepository(client))
+            if (isDotCom || isEnterprise)
             {
                 var hostAddress = HostAddress.Create(repository.CloneUrl);
 
+                notGitHubRepo = false;
+
                 Connection = await connectionManager.GetConnection(hostAddress);
+                Connection?.WhenAnyValue(
+                    x => x.IsLoggedIn,
+                    x => x.IsLoggingIn,
+                    (_, __) => Unit.Default)
+                    .Skip(1)
+                    .Throttle(TimeSpan.FromMilliseconds(100))
+                    .ObserveOn(RxApp.MainThreadScheduler)
+                    .Subscribe(_ => UpdateContent(LocalRepository).Forget());
 
                 if (Connection?.IsLoggedIn == true)
                 {
-                    log.Debug("Found a GitHub repository: {CloneUrl}", repository?.CloneUrl);
-                    Content = navigator;
-                    await ShowDefaultPage();
+                    if (await IsValidRepository(client) == true)
+                    {
+                        log.Debug("Found a GitHub repository: {CloneUrl}", repository?.CloneUrl);
+                        Content = navigator;
+                        await ShowDefaultPage();
+                    }
+                    else
+                    {
+                        notGitHubRepo = true;
+                    }
+                }
+                else if (Connection?.IsLoggingIn == true)
+                {
+                    log.Debug("Found a GitHub repository: {CloneUrl} and logging in", repository?.CloneUrl);
+                    Content = null;
+                }
+                else if (Connection?.ConnectionError != null)
+                {
+                    log.Debug("Found a GitHub repository: {CloneUrl} with login error", repository?.CloneUrl);
+                    loginFailed.Initialize(Connection.ConnectionError.GetUserFriendlyError(ErrorType.LoginFailed));
+                    Content = loginFailed;
                 }
                 else
                 {
                     log.Debug("Found a a GitHub repository but not logged in: {CloneUrl}", repository?.CloneUrl);
                     Content = loggedOut;
                 }
             }
-            else
+            
+            if (notGitHubRepo)
```

Comment:
`else if`?

Reply:
No, this flag is set in the preceding block.